thoth/common/openshift.py

```diff
@@ -687,9 +682,10 @@ def get_solver_names(self) -> list:
                 "Infra namespace is required in order to list solvers"
             )
 
-        response = self.ocp_client.resources.get(api_version="v1", kind="Template").get(
-            namespace=self.infra_namespace, label_selector="template=solver"
-        )
+        # TODO can we refactor this to use _get_template() ?
```

Comment:
Maybe introduce `_get_templates()` instead and return template listing. Additional parameter (something like, `sanitize_only_one=True`) would check if there is just one. A check for != 0 would be performed always. I don't insist on this change as it's internal logic which would not introduce tech-dept elsewhere , just suggestion to next implementation.